app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java

```diff
@@ -355,7 +355,11 @@ public static int getNovaGroupDrawable(@Nullable String novaGroup) {
     }
 
     public static int getSmallImageGrade(Product product) {
-        return getSmallImageGrade(product == null ? null : product.getNutritionGradeFr());
+        String nutritionGrade = null;
+        if (product != null) {
+            nutritionGrade = product.getNutritionGradeFr() != null ? product.getNutritionGradeFr() : product.getNutritionGradeTag();
+        }
+        return getSmallImageGrade(nutritionGrade);
```

Comment:
Can we inline this return? I prefer having multiple exit points than having to remember the value of `nutritionGrade`.

Reply:
Done